src/helpers/getPeoplesWithParents.ts

```diff
@@ -0,0 +1,15 @@
+import { Person } from '../types';
+import { getPersonParent } from './getPersonParent';
+
+export const getPeoplesWithParents = (peopleData: Person[]) => (
+  peopleData.map(person => {
+    const father = getPersonParent(person.fatherName, peopleData);
+    const mother = getPersonParent(person.motherName, peopleData);
```

Comment:
The right decision, but now you have two searches in the middle of another search, which is algorithmically not optimal.
To improve the algorithm, you can create a separate map (`peopleMap`) and add all the people to it using their name as a key (`peopleMap.set(person.name, person)`), after that go through the `people` array and find the person's father and mother (`peopleMap.get(person.motherName)`, `peopleMap.get(person.fatherName)`).
This is a more complicated approach, but more correct on real projects.